### ISSUE TITLE
wishlist: fixture/decorator to facilicate testing on some datasets

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1603]
it would be a nice feature for developers to be able to specify that some test needs a dataset and particular files (defined by globs and/or metadata) for a test, e.g.

```python

@datalad_dataset('///openfmri/ds000001', globs='*/anat')
def test_on_openfmri_dataset(dspath):
   ...
```
or alike which would install/get somewhere and then next time resuses it... so some kinda of a fixture

inspired by @adelavega

**Comment by yarikoptic** [comment]
we have now `@cached_dataset` implemented in #4608 . It has `paths` argument (no globs iirc) to get needed paths.  I will close for now and let's reopen with a new one whenever real use case demands globs or regex support

**Comment by adelavega** [comment]
yay, i have no idea how i inspired this though haha
